mlflow/server/__init__.py

```diff
@@ -84,7 +84,7 @@ def serve_get_metric_history_bulk():
 
 
 # Serve the "experiments/search-datasets" route.
-@app.route(_add_static_prefix("/ajax-api/2.0/mlflow/experiments/search-datasets"))
+@app.route(_add_static_prefix("/ajax-api/2.0/mlflow/experiments/search-datasets"), methods=["POST"])
```

Comment:
fixes bug in https://github.com/mlflow/mlflow/pull/8739. should have been POST all along